client/src/pages/EditProfile.js

```diff
@@ -0,0 +1,107 @@
+import { useForm } from "react-hook-form";
+import { connect } from "react-redux";
+import styled from "styled-components";
+import FormInput from "../components/Input/FormInput";
+import ProfilePic from "../components/Picture/ProfilePic";
+import SubmitButton from "../components/Button/SubmitButton";
+import Title from "../components/Title/Title";
+// dummy data props,context, redux etc
+
+const selfIntroduction =
+  "Lorem ipsum dolor sit amet, consectetuer adipiscing elit";
+const editProfile = true;
+
+const ChangePicButton = styled.div`
+  color: #425af2;
+  margin-bottom: 3rem;
+  text-align: center;
+`;
+
+function getInitials(firstName, lastName) {
+  // function to get the initials given firstname and last name
+  return firstName.charAt(0).toUpperCase() + lastName.charAt(0).toUpperCase();
+}
+
+function EditProfile(props) {
+  // dummy data props,context, redux etc
+  const {
+    firstName,
+    lastName,
+    facebookURL,
+    twitterURL,
+    githubURL,
+    linkedinURL,
+    personalURL,
+    about,
+  } = props.user;
+  const { register, handleSubmit } = useForm();
+
+  const onSubmit = (data) => {
+    console.log(data);
+    // make a put/patch request to backend to update users profile information
+  };
+
+  const labelVariableValue = {
+    // label name, variable name, value
+    "Facebook URL": ["facebookURL", facebookURL],
+    "LinkedIn URL": ["linkedinURL", linkedinURL],
+    "Twitter URL": ["twitterURL", twitterURL],
+    "Github URL": ["githubURL", githubURL],
+    "Personal Website": ["personalURL", personalURL],
+  };
+
+  const renderFormInputs = () => {
+    // iterate and create input
+    return Object.entries(labelVariableValue).map(([key, value]) => {
+      return (
+        <FormInput
+          inputTitle={key}
+          name={value[0]}
+          defaultValue={value[1]}
+          reference={register}
+          key={key}
+        />
+      );
+    });
+  };
+
+  return (
+    <>
+      <Title
+        title={editProfile ? "Edit Profile" : "Complete Profile"}
+        style={{
+          marginBottom: "3rem",
+          marginTop: "2rem",
+          fontWeight: "bolder",
+        }}
+      />
+      <ProfilePic noPic={true} initials={getInitials(firstName, lastName)} />
+      <ChangePicButton>Change</ChangePicButton>
+      <form style={{ display: "flex", flexDirection: "column" }}>
+        <FormInput
+          inputTitle="Self-introduction"
+          name="about"
+          defaultValue={about}
+          reference={register({ maxLength: 160 })}
+        />
+        {renderFormInputs()}
+        <SubmitButton
+          title={"Save Changes"}
+          style={{ marginTop: "1rem", marginBottom: "3rem" }}
+          onClick={handleSubmit(onSubmit)}
+        >
+          Save Changes
+        </SubmitButton>
+      </form>
+    </>
+  );
+}
+
+const mapStateToProps = (state) => {
+  return {
+    user: state.user,
+  };
+};
+
+export default connect(mapStateToProps)(EditProfile);
```

Comment:
@robinv85 Will this be fine for now until we have a way for testing HOCs?

Reply:
I'd say we don't have to worry about that at the moment.
A convention that is sometimes used is to export both the component and the connected component, maybe one named and the other default... 
We could use a naming convention in that case such as "Component" and "ComponentContainer"  ... because otherwise it's not transparent to know what is exported from a file

This presentational / container component pattern was pretty much standard for redux apps for a few years but nowadays with hooks many project don't use redux... as you see here in the recently updated article https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0

If we look at Redux docs they still suggest to use separate folders like `/components` and  `/containers` https://redux.js.org/basics/usage-with-react#designing-component-hierarchy